src/engine/SCons/Environment.py

```diff
@@ -2305,7 +2305,26 @@ def __init__(self, subject, overrides={}):
 
     # Methods that make this class act like a proxy.
     def __getattr__(self, name):
-        return getattr(self.__dict__['__subject'], name)
+        attr = getattr(self.__dict__['__subject'], name)
+        # Here we check if attr is one of the Wrapper classes. For
+        # example when a pseudo-builder is being called from an
+        # OverrideEnvironment.
+        #
+        # These wrappers when they're constructed capture the
+        # Environment they are being constructed with and so will not
+        # have access to overrided values. So we rebuild them with the
+        # OverrideEnvironment so they have access to overrided values.
+        if isinstance(attr, BuilderWrapper):
```

Comment:
I think you can use the clone(self) on BuilderWrapper and MethodWrapper? Seems like it'd be simpler?

Reply:
Done.